### PR TITLE
[DO NOT MERGE] landing a/b

### PR DESCRIPTION
**Pull request by raymondng**
Added simple A/B landing based on the assumption that the files will be named "index1" and "index2"

Don't merge. Want to send this out first because I have been losing internet connectivity regularly these days.

For hashing I picked murmur3 but feel free to suggest other alternatives. For our simple case probably doesn't matter too much -- might even be an overkill. Will monitor. Also I chose not to convert to `InetAddress` (just hash the string) to avoid overhead.

Tested locally. No unit tests yet. I will look into adding one as well.

@andrewconner @eishay

### CODE REVIEW

**Comment by eishay** [comment]
its `index.1.html` and `index.2.html`

**Comment by raymondng** [comment]
Oh will fix. Thanks.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala [hunk 1]
```diff
@@ -34,8 +39,42 @@ object MarketingSiteRouter extends AssetsBuilder with Controller with Logging {
...
+        val ip = Try {
+          request.remoteAddress // remoteAddress looks up 'X-Forwarded-For'
+        } recover {
+          case t: Throwable =>
+            InetAddress.getLocalHost.toString // with affinity this might be sufficient
+        } get
+        val hasher = hashFunction.newHasher()
+        val hc = hasher.putString(ip, Charsets.UTF_8).hash()
+        val hash = (Math.abs(hc.asInt()) % 100) // rough
+        val winner = if (hash < 50) "index1" else "index2"
```
Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala]: wow, that's hard core :-)
thought that `ip.hashCode` is enough

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala [hunk 2]
```diff
@@ -34,8 +39,42 @@ object MarketingSiteRouter extends AssetsBuilder with Controller with Logging {
...
+        log.error(s"[landing] Caught exception $t while parsing queryParam(v):${request.queryString("v")}")
+        None
+    } get
+
+    pickOpt match {
+      case Some(idx) if (idx == 1 || idx == 2) =>
+        "index." + idx
+      case None =>
+        val ip = Try {
+          request.remoteAddress // remoteAddress looks up 'X-Forwarded-For'
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala]: Is this known to throw exceptions?

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala [hunk 3]
```diff
@@ -34,8 +39,42 @@ object MarketingSiteRouter extends AssetsBuilder with Controller with Logging {
...
+      request.getQueryString("v").map(_.toInt)
+    } recover {
+      case t: Throwable =>
+        log.error(s"[landing] Caught exception $t while parsing queryParam(v):${request.queryString("v")}")
+        None
+    } get
+
+    pickOpt match {
+      case Some(idx) if (idx == 1 || idx == 2) =>
+        "index" + idx
```
Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala]:
``` scala
 "index." + idx
```

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala (hunk 2 above)
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala]: Oh good catch. I was using `InetAddress` so I wanted to be on the safe side. Now there is no need.

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala (hunk 1 above)
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala]: Not really :) I started out with `hashCode` and in fact that probably should work fine. Bouncing JVM might be an issue though.

Comment (eishay) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala]: HashCode is using the string chars, it's consistent across vms

Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/MarketingSiteRouter.scala]: Yes you're right. My comment applies in general but does not apply to `String`, as it's a special case.

**Comment by joshm1** [comment]
tested on dev with latest marketing pages. lgtm! thanks!

**Comment by raymondng** [comment]
Cool. Thanks!

**Pull request merged**